File: training-data/julia/temperature_convert.jl
abstract type TempUnit end
struct Celsius <: TempUnit end
struct Fahrenheit <: TempUnit end
struct Kelvin <: TempUnit end

to_celsius(v, ::Celsius) = v
to_celsius(v, ::Fahrenheit) = (v - 32) * 5 / 9
to_celsius(v, ::Kelvin) = v - 273.15

from_celsius(v, ::Celsius) = v
from_celsius(v, ::Fahrenheit) = v * 9 / 5 + 32
from_celsius(v, ::Kelvin) = v + 273.15

convert_temp(v, from::TempUnit, to::TempUnit) = from_celsius(to_celsius(v, from), to)

println(round(convert_temp(100, Celsius(), Fahrenheit()), digits = 2))
println(round(convert_temp(212, Fahrenheit(), Kelvin()), digits = 2))

File: training-data/csharp/TemperatureConvert.cs
using System;

public enum TempUnit { Celsius, Fahrenheit, Kelvin }

public static class TemperatureConvert
{
    static double ToCelsius(double v, TempUnit u) => u switch
    {
        TempUnit.Fahrenheit => (v - 32) * 5 / 9,
        TempUnit.Kelvin => v - 273.15,
        _ => v,
    };

    static double FromCelsius(double v, TempUnit u) => u switch
    {
        TempUnit.Fahrenheit => v * 9 / 5 + 32,
        TempUnit.Kelvin => v + 273.15,
        _ => v,
    };

    public static double Convert(double v, TempUnit from, TempUnit to) => FromCelsius(ToCelsius(v, from), to);

    public static void Main()
    {
        Console.WriteLine($"{Convert(100, TempUnit.Celsius, TempUnit.Fahrenheit):F2}");
        Console.WriteLine($"{Convert(212, TempUnit.Fahrenheit, TempUnit.Kelvin):F2}");
    }
}
